### PR TITLE
Mini Cart Contents block: implement view switcher in editor

### LINKED ISSUE #4857: Mini Cart Contents block: implement view switcher in editor
Similarly to the Cart block, we need to allow switching between the Empty cart and Full cart view in the Mini Cart Contents block when in the editor.

Ideally, we will reuse the switcher which is currently in use in the Cart block:

imatge

### PR DESCRIPTION
**Pull request by dinhtungdu**
This PR adds a view switcher to the Mini Cart Contents block followed the Cart block method. Note that only switcher and placeholder for empty/filled state for editor were added. The front end of the mini cart content is unchanged.

Fixes #4857

### Screenshots
image

#### Automated Tests
* [ ] Changes in this PR are covered by Automated Tests.
  * [ ] Unit tests
  * [ ] E2E tests
### Manual Testing

How to test the changes in this Pull Request:

1. Edit the Mini Cart template part.
2. Select the Mini Cart Contents block.
3. See the switcher in the block toolbar.
4. Try toggling the state, see the content change respectively.

### User Facing Testing
These are steps for user testing (where "user" is someone interacting with this change that is not editing any code).
* [x] Same as above, or
* [ ] See steps below.

### CODE REVIEW

**Review by Aljullu — APPROVED** [review]
Works great and code changes look good.

File: assets/js/blocks/cart-checkout/mini-cart-contents/edit.tsx [hunk 1]
```diff
@@ -2,16 +2,82 @@
...
+			</EditorProvider>
 		</div>
 	);
 };
 
 export default Edit;
+
+export const Save = (): JSX.Element => {
+	return (
+		<div { ...useBlockProps.save() }>
```
Comment (gigitux) [inline assets/js/blocks/cart-checkout/mini-cart-contents/edit.tsx:79]: With the addition of the div, the template no longer matches with the block. (more details on `p1638810187484200?thread_ts=1638803183.465500&cid=C02FL3X7KR6-slack-C02FL3X7KR6`)

**Review by gigitux — CHANGES_REQUESTED** [review]

File: assets/js/blocks/cart-checkout/mini-cart-contents/edit.tsx (hunk 1 above)
Comment (dinhtungdu) [reply assets/js/blocks/cart-checkout/mini-cart-contents/edit.tsx:79]: @gigitux I update the template file, please have another look at this : ).

Comment (gigitux) [reply assets/js/blocks/cart-checkout/mini-cart-contents/edit.tsx:79]: Great! Thanks for the update!

**Review by gigitux — APPROVED** [review]

**Pull request merged**